packages/snaps-controllers/src/services/iframe/test/index.html

```diff
@@ -11,26 +11,35 @@
     <script>
       (function () {
         try {
-          const snap = top.querySelector('iframe[data-testid="snaps-iframe"]');
+          // Parent is the iframe created by Vitest containing the Snap iframe
+          // and this page as a sibling iframe. We try to access the Snap iframe
+          // from the parent iframe.
+          const snap = parent.document.querySelector(
+            'iframe[data-testid="snaps-iframe"]',
+          );
           if (!snap) {
-            window.parent.postMessage('Failed to find snap iframe.', '*');
+            window.parent.postMessage('Failed to find Snap iframe.', '*');
             return;
           }
 
           // We try to access the snap iframe document. If this works, it means
           // that the iframe is not sandboxed. Otherwise it will throw an error.
-          snap.document;
-
-          window.parent.postMessage(
-            'Same origin frames access worked which might suggest sandboxing failed.',
-            '*',
-          );
```

Comment:
This test was wrong in the first place. The iframe was served from a different origin, and trying to access `top.querySelector` resulted in:

> SecurityError: Failed to read a named property 'querySelector' from 'Window': Blocked a frame with origin "http://localhost:4567/" from accessing a cross-origin frame.

So the `snap.document` was never reached, and `top.querySelector` isn't even a function even if it was accessible. 😅

I confirmed that the new test properly tests the sandboxing behaviour, as it's now served from the same origin. Removing the sandbox property from the iframe now results in a test failure.